Backend/templates/gitlab-ci/master-gitlab-ci.yml

```diff
@@ -0,0 +1,274 @@
+    - docker build -t $CI_REGISTRY/$CI_PROJECT_PATH/builds:$BUILD_NUMBER .
+    - docker push $CI_REGISTRY/$CI_PROJECT_PATH/builds:$BUILD_NUMBER
+
+
+TagBuildNumber:
+  stage: TagBuildNumber
+  image: samueldebruyn/debian-git
+  only:
+    - master
+  script:
+    - BUILD_NUMBER=`cat .build_number`
+    - echo "Build number $BUILD_NUMBER"
+    
+    - git tag builds/$BUILD_NUMBER
+    - git remote set-url origin https://gitlab-ci-token:$GL_TOKEN@gitlab.com/$CI_PROJECT_PATH.git/
+    - git push origin --tags
+
+
+TagVersion:
+  stage: TagVersion
+  image: registry.gitlab.com/juhani/go-semrel-gitlab:$GSG_DOCKER_TAG
+  when: manual
+  only:
+    - master
+  script:
+    - release changelog
+    - release commit-and-tag CHANGELOG.md --create-tag-pipeline
+
+
+
+##################
+# Build tag jobs #
+##################
+
+DeployVersionDev: 
+  stage: BuildNumber
+  only:
+    - /builds/
+  except:
+    - branches
+  script:
+    - echo $CI_COMMIT_TAG | sed -n 's/builds\/\([0-9]\)/\1/p' > .build_number
+    - echo "Build number "$(cat .build_number)
+  artifacts:
+    paths:
+      #TODO maybe this should be a different name? 🤔
+      - .build_number 
+
+
+DeployDev:
+  extends: .deploy_template
+  variables:
+    PROJECT: "youclap-dev"
+  only:
+    - /builds/
+
+
+DocumentationDev:
+  extends: .documentation_template
+  only:
+    - /builds/
+  dependencies:
+    - DeployVersionDev 
+
+
+DeployDocumentationDev:
+  extends: .deploy_documentation_template
+  variables:
+    PROJECT: "youclap-dev"
+    DOC: "dev" #TODO find a better name
+  only:
+    - /builds/
+  dependencies:
+    - DeployVersionDev
+    - DocumentationDev
+
+
+#####################
+# Releases tag jobs #
+#####################
+
+DeployVersionProd: 
+  stage: BuildNumber
+  only:
+    - /releases/
+  except:
+    - branches
+  script:
+    - echo $CI_COMMIT_TAG | sed -n 's/releases\/\([0-9].[0-9].[0-9]\)/\1/p' > .build_number
+    - cat .build_number
+  artifacts:
+    paths:
+      - .build_number
+
+
+DeployProd:
+  extends: .deploy_template
+  only:
+    - /releases/
+  variables:
+    PROJECT: "youclap-prod"
+
+
+DocumentationProd:
+  extends: .documentation_template
+  only:
+    - /releases/
+  dependencies:
+    - DeployVersionProd
+
+
+DeployDocumentationProd:
+  extends: .deploy_documentation_template
+  variables:
+    PROJECT: "youclap-prod"
+    DOC: "prod"
+  only:
+    - /releases/
+  dependencies:
+    - DeployVersionProd
+    - DocumentationProd
+
+
+#############
+# Templates #
+#############
+
+.build_template:
+  stage: Build
+  only:
+    - master
+
+
+.deploy_template:
+  stage: Deploy
+  image: devth/helm:$HELM_DOCKER_TAG
+  except:
+    - branches
+  script:
+    - BUILD_NUMBER=`cat .build_number`
+    - echo "Build number $BUILD_NUMBER"
+
+    - gcloud auth activate-service-account --key-file=$GCLOUD_SERVICE_ACCOUNT_WS_KUBERNETES --project $PROJECT
+    - gcloud container clusters get-credentials gcloud-cluster-dev --zone europe-west1-d --project $PROJECT
```

Comment:
hum i think the cluster and zone should be environment variables, this can change quickly

Reply:
dune to